drizzle-kit/tests/introspect/pg.test.ts

```diff
@@ -892,3 +894,46 @@ test('multiple policies with roles from schema', async () => {
 	expect(statements.length).toBe(0);
 	expect(sqlStatements.length).toBe(0);
 });
+
+
+
+test('introspect foreign keys', async () => {
+	const client = new PGlite();
+
+	const userProfileImages = pgTable('user_profile_images', {
+		id: uuid('id').primaryKey(),
+		ownerId: uuid('owner_id').notNull(),
+		image_blob: text('image_blob').notNull(), // whatever
+	}, (table) => [
+		// Ensures (id, ownerId) is also unique for the composite fk to work
+		unique().on(table.id, table.ownerId)
+	]);
+
+	const userProfileCoverImages = pgTable('user_profile_cover_images', {
+		id: uuid('id').primaryKey(),
+		ownerId: uuid('owner_id').notNull(),
+		profileImageId: uuid('profile_image_id')
+			.notNull(),
+	}, (table) => [
+		// Guarantees each cover image matches the correct profile + owner
+		foreignKey({
+			columns: [table.ownerId, table.profileImageId],
+			foreignColumns: [userProfileImages.ownerId, userProfileImages.id],
```

Comment:
The order of these matters. Previously it had weird behaviour where it would appear that these were sorted. The information is in the linked issue. Perhaps there is a better way to describe the preconditions